### PR TITLE
Adding pvc waiter for bound and statefulset replicas running

### PR DESCRIPTION
**Pull request by paigerube14**
### Description
Adding more object types to wait for bound/running
Includes: 
PVC
Statefulsets to have all running pods

### Fixes
Be able to properly wait for creation of multiple objects

### CODE REVIEW

**Comment by paigerube14** [comment]
@rsevilla87 @jtaleric PTAL when you get a chance, thanks!

File: pkg/burner/waiters.go [hunk 1]
```diff
@@ -67,6 +69,8 @@ func (ex *Executor) waitForObjects(ns string) {
 				go waitForVMIRS(ns, ex.Config.MaxWaitTimeout, &wg)
 			case "Job":
 				go waitForJob(ns, ex.Config.MaxWaitTimeout, &wg)
+			case "PVC":
```
Comment (rsevilla87) [inline pkg/burner/waiters.go:72]: Is this kind correct? The kind of the PVC is actually PersistentVoumeClaim according to api-resources

```
rsevilla@wonderland ~/labs/kube-burner (openshift-wrapper) $ oc api-resources | grep pvc
persistentvolumeclaims                pvc                 v1                                            true         PersistentVolumeClaim
```

File: pkg/burner/waiters.go [hunk 2]
```diff
@@ -111,6 +115,34 @@ func waitForRS(ns string, maxWaitTimeout time.Duration, wg *sync.WaitGroup) {
...
+func waitForStatefulSet(ns string, maxWaitTimeout time.Duration, wg *sync.WaitGroup) {
+	defer wg.Done()
+	wait.PollImmediate(1*time.Second, maxWaitTimeout, func() (bool, error) {
+		rss, err := ClientSet.AppsV1().StatefulSets(ns).List(context.TODO(), metav1.ListOptions{})
+		if err != nil {
+			return false, err
+		}
+		for _, rs := range rss.Items {
+			if *rs.Spec.Replicas != rs.Status.ReadyReplicas {
+				log.Debugf("Waiting for replicas from replicaSets in ns %s to be ready", ns)
```
Comment (rsevilla87) [inline pkg/burner/waiters.go:127]:
```suggestion
				log.Debugf("Waiting for replicas from statefulSets in ns %s to be ready", ns)
```

File: pkg/burner/waiters.go [hunk 3]
```diff
@@ -111,6 +115,34 @@ func waitForRS(ns string, maxWaitTimeout time.Duration, wg *sync.WaitGroup) {
 	})
 }
 
+func waitForStatefulSet(ns string, maxWaitTimeout time.Duration, wg *sync.WaitGroup) {
+	defer wg.Done()
+	wait.PollImmediate(1*time.Second, maxWaitTimeout, func() (bool, error) {
+		rss, err := ClientSet.AppsV1().StatefulSets(ns).List(context.TODO(), metav1.ListOptions{})
```
Comment (rsevilla87) [inline pkg/burner/waiters.go:121]: For convention, can you rename this variable to stss?

File: pkg/burner/waiters.go [hunk 4]
```diff
@@ -111,6 +115,34 @@ func waitForRS(ns string, maxWaitTimeout time.Duration, wg *sync.WaitGroup) {
...
+		if err != nil {
+			return false, err
+		}
+		for _, rs := range rss.Items {
```
Comment (rsevilla87) [inline pkg/burner/waiters.go:125]: Same, s/rs/sts

File: pkg/burner/waiters.go (hunk 1 above)
Comment (paigerube14) [reply pkg/burner/waiters.go]: Updated to use PersistentVolumeClaim as type

**Comment by paigerube14** [comment]
@rsevilla87 Some output from one of my runs where I used my built image to verify the PVC objects were bound before moving on (not showing a whole lot with the additions I added)

```
+ bash run.sh
###############################################
Workload: custom
Workload template: /Users/prubenda/PycharmProjects/svt/perfscale_regression_ci/scripts/scalability/descheduler/../../../kubeburner-object-templates/descheduler-local-storage.yml
Metrics profile: metrics-profiles/metrics.yaml
Alerts profile: 
QPS: 20
Burst: 20
UUID: 3433626D-E258-40D6-9247-D1807671BF8A
Job iterations: 
###############################################
Mon Nov 14 16:50:56 UTC 2022 Indexing enabled, using metrics from metrics-profiles/metrics.yaml
~/PycharmProjects/svt/perfscale_regression_ci/kubeburner-object-templates ~/PycharmProjects/svt/perfscale_regression_ci/scripts/scalability/descheduler/e2e-benchmarking/workloads/kube-burner
INFO[2022-11-14 11:50:56] 🔥 Starting kube-burner (pvc@63c3f286d139d5ef89e0433b7c0b11a1d7b1eb7d) with UUID 3433626D-E258-40D6-9247-D1807671BF8A 
INFO[2022-11-14 11:50:57] 📈 Creating measurement factory               
INFO[2022-11-14 11:50:57] Preparing create job: local-storage-desched  
INFO[2022-11-14 11:50:57] Job local-storage-desched: 1 iterations with 1 PersistentVolumeClaim replicas 
INFO[2022-11-14 11:50:57] Job local-storage-desched: 1 iterations with 1 ReplicationController replicas 
INFO[2022-11-14 11:50:57] Job local-storage-desched: 1 iterations with 1 ReplicationController replicas 
INFO[2022-11-14 11:50:57] Pre-load: images from job local-storage-desched 
INFO[2022-11-14 11:50:57] Pre-load: Sleeping for 1m0s                  
INFO[2022-11-14 11:51:57] Pre-load: Deleting namespace preload-kube-burner 
INFO[2022-11-14 11:51:57] Deleting namespaces with label kube-burner-preload=true 
INFO[2022-11-14 11:51:57] Waiting for namespaces to be definitely deleted 
INFO[2022-11-14 11:52:05] Triggering job: local-storage-desched        
INFO[2022-11-14 11:52:05] QPS: 20                                      
INFO[2022-11-14 11:52:05] Burst: 20                                    
INFO[2022-11-14 11:52:05] Running job local-storage-desched            
INFO[2022-11-14 11:52:05] Waiting up to 3h0m0s all job actions to be completed 
INFO[2022-11-14 11:53:20] Actions in namespace local-storage-desched-1 completed 
INFO[2022-11-14 11:53:20] Finished the create job in 76 seconds        
INFO[2022-11-14 11:53:20] Verifying created objects                    
INFO[2022-11-14 11:53:20] persistentvolumeclaims found: 1 Expected: 1  
INFO[2022-11-14 11:53:20] replicationcontrollers found: 1 Expected: 1  
INFO[2022-11-14 11:53:20] replicationcontrollers found: 1 Expected: 1  
INFO[2022-11-14 11:53:20] Job local-storage-desched took 75.81 seconds 
INFO[2022-11-14 11:53:20] Finished execution with UUID: 3433626D-E258-40D6-9247-D1807671BF8A 
INFO[2022-11-14 11:53:20] 👋 Exiting kube-burner                        
~/PycharmProjects/svt/perfscale_regression_ci/scripts/scalability/descheduler/e2e-benchmarking/workloads/kube-burner
Mon Nov 14 16:53:28 UTC 2022 Indexing benchmark metadata to https://search-perfscale-dev-chmf5l4sh66lvxbnadi4bznl3a.us-west-2.es.amazonaws.com:443/ripsaw-kube-burner
```

**Comment by rsevilla87** [comment]
@paigerube14 , I ran some satisfatory tests with StatefulSets as well. (I modified the kubelet-density pod by a statefulset)
```
 $ ../../../bin/amd64/kube-burner init -c kubelet-density.yml  --log-level=debug                                                                                                                                                              
INFO[2022-11-15 12:04:56] 🔥 Starting kube-burner (pvc@63c3f286d139d5ef89e0433b7c0b11a1d7b1eb7d) with UUID 94e023a3-d83d-444a-8f35-4fa39f69cc57                                                                                                                                                                              
INFO[2022-11-15 12:04:56] 📈 Creating measurement factory                                                                                                                                                                                                                                                                    
INFO[2022-11-15 12:04:56] Registered measurement: podLatency                                                                                                                                                                                                                                                                 
INFO[2022-11-15 12:04:56] Preparing create job: kubelet-density                                                                                                                                                                                                                                                              
DEBU[2022-11-15 12:04:56] Processing template: templates/pod.yml                                                                                                                                                                                                                                                             
INFO[2022-11-15 12:04:56] Job kubelet-density: 1 iterations with 1 StatefulSet replicas                                                                                                                                                                                                                                      
INFO[2022-11-15 12:04:56] Triggering job: kubelet-density                                                                                                                                                                                                                                                                    
INFO[2022-11-15 12:05:07] Creating Pod latency watcher for kubelet-density                                                              
INFO[2022-11-15 12:05:07] QPS: 20                                                                                                       
INFO[2022-11-15 12:05:07] Burst: 20                                                                                                     
INFO[2022-11-15 12:05:07] Running job kubelet-density                                                                                   
DEBU[2022-11-15 12:05:07] Created namespace: kubelet-density                                                                            
DEBU[2022-11-15 12:05:07] Creating object replicas from iteration 1                                                                     
DEBU[2022-11-15 12:05:07] Created StatefulSet/kubelet-density-1 in namespace kubelet-density                                            
INFO[2022-11-15 12:05:07] Waiting up to 3h0m0s for actions to be completed                                                              
DEBU[2022-11-15 12:05:07] Waiting for replicas from statefulSets in ns kubelet-density to be ready                                      
DEBU[2022-11-15 12:05:08] Waiting for replicas from statefulSets in ns kubelet-density to be ready                                      
DEBU[2022-11-15 12:05:09] Pod kubelet-density-1-0 is ready                                                                              
DEBU[2022-11-15 12:05:09] Waiting for replicas from statefulSets in ns kubelet-density to be ready                                      
DEBU[2022-11-15 12:05:10] Waiting for replicas from statefulSets in ns kubelet-density to be ready                                      
DEBU[2022-11-15 12:05:11] Waiting for replicas from statefulSets in ns kubelet-density to be ready                                      
DEBU[2022-11-15 12:05:12] Waiting for replicas from statefulSets in ns kubelet-density to be ready                                      
DEBU[2022-11-15 12:05:13] Waiting for replicas from statefulSets in ns kubelet-density to be ready                                      
DEBU[2022-11-15 12:05:14] Waiting for replicas from statefulSets in ns kubelet-density to be ready                                      
DEBU[2022-11-15 12:05:15] Waiting for replicas from statefulSets in ns kubelet-density to be ready                                      
DEBU[2022-11-15 12:05:16] Waiting for replicas from statefulSets in ns kubelet-density to be ready                                      
DEBU[2022-11-15 12:05:17] Waiting for replicas from statefulSets in ns kubelet-density to be ready                                      
DEBU[2022-11-15 12:05:18] Waiting for replicas from statefulSets in ns kubelet-density to be ready                                      
DEBU[2022-11-15 12:05:19] Waiting for replicas from statefulSets in ns kubelet-density to be ready                                      
DEBU[2022-11-15 12:05:20] Waiting for replicas from statefulSets in ns kubelet-density to be ready                                      
DEBU[2022-11-15 12:05:21] Waiting for replicas from statefulSets in ns kubelet-density to be ready                                      
DEBU[2022-11-15 12:05:22] Waiting for replicas from statefulSets in ns kubelet-density to be ready                                      
DEBU[2022-11-15 12:05:23] Waiting for replicas from statefulSets in ns kubelet-density to be ready                                      
DEBU[2022-11-15 12:05:24] Waiting for replicas from statefulSets in ns kubelet-density to be ready                                      
DEBU[2022-11-15 12:05:25] Waiting for replicas from statefulSets in ns kubelet-density to be ready                                                            
DEBU[2022-11-15 12:05:26] Waiting for replicas from statefulSets in ns kubelet-density to be ready                                                            
DEBU[2022-11-15 12:05:27] Waiting for replicas from statefulSets in ns kubelet-density to be ready                                                            
DEBU[2022-11-15 12:05:28] Waiting for replicas from statefulSets in ns kubelet-density to be ready                                                            
DEBU[2022-11-15 12:05:29] Pod kubelet-density-1-1 is ready                                                                                                    
DEBU[2022-11-15 12:05:29] Waiting for replicas from statefulSets in ns kubelet-density to be ready                                                            
DEBU[2022-11-15 12:05:30] Waiting for replicas from statefulSets in ns kubelet-density to be ready                                                            
DEBU[2022-11-15 12:05:31] Waiting for replicas from statefulSets in ns kubelet-density to be ready                                                            
DEBU[2022-11-15 12:05:32] Waiting for replicas from statefulSets in ns kubelet-density to be ready                                                            
DEBU[2022-11-15 12:05:33] Waiting for replicas from statefulSets in ns kubelet-density to be ready                                                            
DEBU[2022-11-15 12:05:34] Waiting for replicas from statefulSets in ns kubelet-density to be ready                                                            
DEBU[2022-11-15 12:05:35] Waiting for replicas from statefulSets in ns kubelet-density to be ready                                                            
DEBU[2022-11-15 12:05:36] Waiting for replicas from statefulSets in ns kubelet-density to be ready                                                            
DEBU[2022-11-15 12:05:37] Waiting for replicas from statefulSets in ns kubelet-density to be ready                                                            
DEBU[2022-11-15 12:05:38] Waiting for replicas from statefulSets in ns kubelet-density to be ready                                                            
DEBU[2022-11-15 12:05:39] Waiting for replicas from statefulSets in ns kubelet-density to be ready                                                            
DEBU[2022-11-15 12:05:40] Waiting for replicas from statefulSets in ns kubelet-density to be ready                                                            
DEBU[2022-11-15 12:05:41] Waiting for replicas from statefulSets in ns kubelet-density to be ready                                                            
DEBU[2022-11-15 12:05:42] Waiting for replicas from statefulSets in ns kubelet-density to be ready                                                            
DEBU[2022-11-15 12:05:43] Waiting for replicas from statefulSets in ns kubelet-density to be ready                                                            
DEBU[2022-11-15 12:05:44] Waiting for replicas from statefulSets in ns kubelet-density to be ready                                                            
DEBU[2022-11-15 12:05:45] Waiting for replicas from statefulSets in ns kubelet-density to be ready                                                            
DEBU[2022-11-15 12:05:46] Waiting for replicas from statefulSets in ns kubelet-density to be ready                                                            
DEBU[2022-11-15 12:05:47] Waiting for replicas from statefulSets in ns kubelet-density to be ready                                                            
DEBU[2022-11-15 12:05:48] Waiting for replicas from statefulSets in ns kubelet-density to be ready                                                            
DEBU[2022-11-15 12:05:49] Pod kubelet-density-1-2 is ready                                                                                                    
INFO[2022-11-15 12:05:49] Actions in namespace kubelet-density completed                                                                                      
INFO[2022-11-15 12:05:49] Finished the create job in 42 seconds                                                                                               
INFO[2022-11-15 12:05:49] Verifying created objects                                                                                                           
INFO[2022-11-15 12:05:49] statefulsets found: 1 Expected: 1                                                                                                   
INFO[2022-11-15 12:05:49] Stopping measurement: podLatency                                                                                                    
INFO[2022-11-15 12:05:49] Evaluating latency thresholds                                                                                                       
INFO[2022-11-15 12:05:49] Writing latency metrics in collected-metrics/kubelet-density-podLatency.json                                                        
INFO[2022-11-15 12:05:49] Writing latency metrics in collected-metrics/kubelet-density-podLatency-summary.json                                                
INFO[2022-11-15 12:05:49] kubelet-density: PodScheduled 50th: 5 99th: 10 max: 10 avg: 5                                                                       
INFO[2022-11-15 12:05:49] kubelet-density: ContainersReady 50th: 1472 99th: 1556 max: 1556 avg: 1488                                                          
INFO[2022-11-15 12:05:49] kubelet-density: Initialized 50th: 25 99th: 31 max: 31 avg: 23                                                                      
INFO[2022-11-15 12:05:49] kubelet-density: Ready 50th: 1472 99th: 1556 max: 1556 avg: 1488                                                                    
INFO[2022-11-15 12:05:49] Job kubelet-density took 53.22 second
```

**Review by rsevilla87 — APPROVED** [review]
LGTM!

**Pull request merged**